tls/s2n_cipher_suites.c

```diff
@@ -1016,19 +1016,24 @@ static int s2n_wire_ciphers_contain(const uint8_t * match, const uint8_t * wire,
  * 1. Certificates that match the client's ServerName extension.
  * 2. Default certificates
  */
-static struct s2n_cert_chain_and_key *s2n_conn_get_compatible_cert_chain_and_key(struct s2n_connection *conn, struct s2n_cipher_suite *cipher_suite)
+struct s2n_cert_chain_and_key *s2n_conn_get_compatible_cert_chain_and_key_by_type(struct s2n_connection *conn, const s2n_authentication_method auth_method)
 {
     if (conn->handshake_params.exact_sni_match_exists) {
         /* This may return NULL if there was an SNI match, but not a match the cipher_suite's authentication type. */
-        return conn->handshake_params.exact_sni_matches[cipher_suite->auth_method];
+        return conn->handshake_params.exact_sni_matches[auth_method];
     } if (conn->handshake_params.wc_sni_match_exists) {
-        return conn->handshake_params.wc_sni_matches[cipher_suite->auth_method];
+        return conn->handshake_params.wc_sni_matches[auth_method];
     } else {
         /* We don't have any name matches. Use the default certificate that works with the key type. */
-        return conn->config->default_cert_per_auth_method.certs[cipher_suite->auth_method];
+        return conn->config->default_cert_per_auth_method.certs[auth_method];
     }
 }
 
+static struct s2n_cert_chain_and_key *s2n_conn_get_compatible_cert_chain_and_key(struct s2n_connection *conn, struct s2n_cipher_suite *cipher_suite)
+{
+    return s2n_conn_get_compatible_cert_chain_and_key_by_type(conn, cipher_suite->auth_method);
+}
+
 static int s2n_set_cipher_and_cert_as_server(struct s2n_connection *conn, uint8_t * wire, uint32_t count, uint32_t cipher_suite_len)
```

Comment:
You integrated this new 1.3 logic without modifying the current cipher selection much at all. Nicely done!